### ISSUE TITLE
improvement request: Add wrapper for Rebar

### ISSUE THREAD

**Issue by eibre** [DynamoDS/DynamoRevit#1689]
Many of the nodes in the [Dynamo For Revit](https://github.com/tt-acm/DynamoForRebar) package can not be used with existing Rebars selected by OOTB Dynamo nodes because they are wrapped as `Revit.Elements.UnknownElement`

The wrapper already exists in the Dynamo For Revit package, if approved by @moethu I would prefer it moved to DynamoRevit.

![image](https://cloud.githubusercontent.com/assets/5968613/26452726/4fab7342-4161-11e7-9f9f-4d10e6e53534.png)

https://github.com/tt-acm/DynamoForRebar/issues/72

**Comment by moethu** [comment]
@eibre agree, this would be nice. Back then when I developed this package it needed to be independent of DynamoRevit, which brought some limitations. Feel free to move the wrappers to DynamoRevit.

**Comment by smangarole** [comment]
@kronz and @mjkkirschner : Ping. 

I think we need your help in making this decision.

**Comment by ksobon** [comment]
@eibre that's a good idea, but...Dynamo for Rebar is a 3rd party package. If we decide to roll that Rebar wrapper into DynamoRevit, great, but then @moethu would still have to update his package to now work with the new wrapper. There are a lot of moving parts here. 

@moethu couldn't you just update your package to allow for UnknownElement input, and then feed it through your internal constructors? I am sure that's a lot of work but maybe it's worth it.

**Comment by moethu** [comment]
@ksobon yes this would be the easiest solution. will do that, should be done quite fast

**Comment by eibre** [comment]
@ksobon @moethu Are you talking about changing the input types from `Revit.Elements.Rebar` to `Revit.Elements.Element`? Is there no other reason to have a wrapper than be able to specify the input type?

If it's preferred to keep most Revit elements as 3rd party packages, shouldn't there be a way for packages to "hook" into DynamoRevit and add wrappers on startup?

Personally, I would still like to have the Rebar wrappers and the _Create_ nodes as a part of DynamoRevit, but I see that it will require some cleanup.

**Comment by ksobon** [comment]
I don't mind having it all be part of DynamoRevit either but that involves a little more work that @moethu would not be getting compensated for. Paging @kronz 

Yes, it would be nice to have ability to extend DynamoRevit classes, but currently that's not exactly possible. This, however, echos what I was saying here already: https://github.com/DynamoDS/Dynamo/issues/7105

**Comment by moethu** [comment]
Agree, extending wrappers of DynamoRevit would be great for anyone writing new packages - but it will be hard to maintain it coordinated because people might create duplicates or similar wrappers for same classes.

Adding Rebar to DynamoRevit wouldn't be the best solution because I'd have to wait until the next DynamoRevit release to update the rebar package and if someone continues using the old rebar package there would be conflicting wrappers.

I think the best solution would be to accept input of type ```Revit.Elements.Element``` and drop the rebar internal wrapper entirely. Later down the line there can still be a rebar wrapper in DynamoRevit.

**Comment by kronz** [comment]
Let me get a technical consult from @mjkkirschner and @ikeough

**Comment by johnpierson** [comment]
Closing this out - the work discussed here was taken on by the package maintainer, and there has been no activity since.

If the gap still exists on a current build, please open a fresh issue.
